Code/Framework/AzFramework/CMakeLists.txt

```diff
@@ -33,6 +32,15 @@ ly_add_target(
             3rdParty::lz4
 )
 
+ly_map_cache_variable_configs(LY_SEARCH_MODE_DEFINE LY_ARCHIVE_FILE_SEARCH_MODE
+    "PREFIX;LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=" "release;2")
+
```

Comment:
this is a bit overcomplicated to do this way, why not just using a generation expression? 
```
set(LY_SEARCH_MODE_DEFINE $<$<CONFIG:release>:LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=2>)
```
That should expand to `LY_ARCHIVE_FILE_SEARCH_MODE_DEFAULT=2` for release and nothing for the other ones

Reply:
Yeah we can do that instead.
I was trying to avoid complications on the user side, as they would need to have knowledge of CMake Generator expressions if they desired to set per config values for the define, but I think that is fair trade off.

If a user wants to have different values per config, they need to have a base level of CMake knowledge